core/src/test/java/org/testcontainers/utility/AuthenticatedImagePullTest.java

```diff
@@ -83,17 +83,77 @@ public void testThatAuthLocatorIsUsed() throws Exception {
 
         // a push will use the auth locator for authentication, although that isn't the goal of this test
         putImageInRegistry();
+    }
+
+    @Before
+    public void removeImageFromLocalDocker() {
+        // remove the image tag from local docker so that it must be pulled before use
+        client.removeImageCmd(testImageNameWithTag).withForce(true).exec();
+    }
 
+    @AfterClass
+    public static void tearDown() {
+        RegistryAuthLocator.setInstance(originalAuthLocatorSingleton);
+    }
+
+    @Test
+    public void testThatAuthLocatorIsUsedForContainerCreation() {
         // actually start a container, which will require an authenticated pull
-        try (final GenericContainer container = new GenericContainer<>(testImageNameWithTag)
+        try (final GenericContainer<?> container = new GenericContainer<>(testImageNameWithTag)
             .withCommand("/bin/sh", "-c", "sleep 10")) {
             container.start();
 
             assertTrue("container started following an authenticated pull", container.isRunning());
         }
     }
 
-    private void putImageInRegistry() throws InterruptedException {
+    @Test
+    public void testThatAuthLocatorIsUsedForDockerfileBuild() throws IOException {
+        // Prepare a simple temporary Dockerfile which requires our custom private image
+        Path tempContext = Files.createTempDirectory(Paths.get("."), this.getClass().getSimpleName() + "-test-");
+        Path tempFile = Files.createTempFile(tempContext, "test", ".Dockerfile");
+        String dockerFileContent = "FROM " + testImageNameWithTag;
+        Files.write(tempFile, dockerFileContent.getBytes());
+
+        // Start a container built from a derived image, which will require an authenticated pull
+        try (final GenericContainer<?> container = new GenericContainer<>(
+            new ImageFromDockerfile()
+                .withDockerfile(tempFile)
+        )
+            .withCommand("/bin/sh", "-c", "sleep 10")) {
+            container.start();
+
+            assertTrue("container started following an authenticated pull", container.isRunning());
+        }
+    }
+
+    @Test
+    public void testThatAuthLocatorIsUsedForDockerComposePull() throws IOException {
+        // Prepare a simple temporary Docker Compose manifest which requires our custom private image
+        Path tempContext = Files.createTempDirectory(Paths.get("."), this.getClass().getSimpleName() + "-test-");
+        Path tempFile = Files.createTempFile(tempContext, "test", ".docker-compose.yml");
+        @Language("yaml") String composeFileContent =
+            "version: '2.0'\n" +
+                "services:\n" +
+                "  privateservice:\n" +
+                "      command: /bin/sh -c 'sleep 60'\n" +
+                "      image: " + testImageNameWithTag;
```

Comment:
I wish we had text blocks!

I would have preferred to just point to a static file on disk, but unfortunately our dockerized private registry has a port number that can vary, and thus the image name changes.

Hence, constructing a tiny YAML file in a string seemed like a tolerable evil.

Reply:
We can, with https://github.com/bsideup/jabel ;)